src/main/java/net/mcreator/generator/ListTemplate.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2022, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.generator;
+
+import java.io.File;
+
+public class ListTemplate extends GeneratorTemplate {
+	private GeneratorTemplatesList templatesList;
+	private final int listItemIndex;
+
+	ListTemplate(File file, String templateIdentificator, Object templateData) {

```

Comment:
Why/where is a template without templatesList and index used? 

To me it seems a bit strange if a list template can exist without both elements that define the list template? Isn't it then a regular template?

Reply:
Could you not assign additional data with a setter instead of making a new list template then?

If this is not possible, then it seems to me like one object is being used for two different purposes, which is not ok